$OpenBSD: patch-src_DownloadQueue_cpp,v 1.1 2011/07/26 15:32:36 dcoppa Exp $

Fix a crash with sources which are not compatible with amule's
encryption settings (commit c204a4fbba4fe7a8ad43fa59b57a9ad57659de15)

--- src/DownloadQueue.cpp.orig	Tue Jul 26 15:04:43 2011
+++ src/DownloadQueue.cpp	Tue Jul 26 15:10:01 2011
@@ -628,7 +628,6 @@ void CDownloadQueue::CheckAndAddKnownSource(CPartFile*
 	// Filter sources which are incompatible with our encryption setting (one requires it, and the other one doesn't supports it)
 	if ( (source->RequiresCryptLayer() && (!thePrefs::IsClientCryptLayerSupported() || !source->HasValidHash())) || (thePrefs::IsClientCryptLayerRequired() && (!source->SupportsCryptLayer() || !source->HasValidHash())))
 	{
-		source->Safe_Delete();
 		return;
 	}		
 	
